insonmnia/hub/server.go

```diff
@@ -476,6 +489,42 @@ func (h *Hub) startTask(ctx context.Context, request *structs.StartTaskRequest)
 	return reply, nil
 }
 
+func (h *Hub) checkImageWhitelist(registry string, image string, auth string) (bool, reference.Named, error) {
```

Comment:
I strictly against making the entire method for a single field `whitelist`. Decompose into `Whitelist` interface.